src/features/chart/redux/__test__/middleware.test.ts

```diff
@@ -0,0 +1,73 @@
+import { chartMiddleware } from "../middleware";
+import { updateChartRangeAction, ChartData } from "../actions";
+import { BOOTSTRAP } from "store";
+
+jest.mock("services/socketService");
+
+describe("When testing chart middleware", () => {
+  let mockSocket: any;
+  let socketService: any;
+  let store: any;
+  let next: jest.Mock;
+
+  beforeEach(() => {
+    const payload = [{ close: 1, date: "1" }];
+    mockSocket = {
+      emit: jest.fn(),
+      on: jest.fn((name: string, callback: (payload: ChartData[]) => void) => {
+        callback(payload);
+      })
+    };
+
+    socketService = {
+      get: () => mockSocket
+    };
+
+    const state = { stockData: { selectedStock: { symbol: "AAPL" } } };
+
+    store = {
+      getState: jest.fn(() => state),
+      dispatch: jest.fn()
+    };
+
+    next = jest.fn();
+  });
+
+  it("it should create a websocket for the chart data on startup", () => {
+    const action = { type: BOOTSTRAP };
+    const middleware = chartMiddleware({ socketService });
+    middleware(store)(next)(action);
+
+    expect(mockSocket.on).toHaveBeenCalledWith("chartData", expect.anything());
```

Comment:
why you did not test the dispatch action within the socket.on?

Reply:
I'd say they're two different functions, one is to test a `socket` creation is called, and the other is to test if `dispatch` is called, so I'm not sure why they'd need to be combined